src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.hpp

```diff
@@ -0,0 +1,317 @@
+    DataType dr_metric_angular_potential{};
+  };
+
+  using equation_of_state_type = EquationsOfState::PolytropicFluid<true>;
+
+  /// The central density of the star.
+  struct CentralDensity {
+    using type = double;
+    static constexpr OptionString help = {"The central density of the star."};
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  /// The polytropic constant of the polytropic fluid.
+  struct PolytropicConstant {
+    using type = double;
+    static constexpr OptionString help = {
+        "The polytropic constant of the fluid."};
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  /// The polytropic exponent of the polytropic fluid.
+  struct PolytropicExponent {
+    using type = double;
+    static constexpr OptionString help = {
+        "The polytropic exponent of the fluid."};
+    static type lower_bound() noexcept { return 1.; }
+  };
+
+  using options =
+      tmpl::list<CentralDensity, PolytropicConstant, PolytropicExponent>;
+
+  static constexpr OptionString help = {
+      "A static, spherically-symmetric star found by solving the \n"
+      "Tolman-Oppenheimer-Volkoff (TOV) equations, with a given central \n"
+      "density and polytropic fluid."};
+
+  TovStar() = default;
+  TovStar(const TovStar& /*rhs*/) = delete;
+  TovStar& operator=(const TovStar& /*rhs*/) = delete;
+  TovStar(TovStar&& /*rhs*/) noexcept = default;
+  TovStar& operator=(TovStar&& /*rhs*/) noexcept = default;
+  ~TovStar() = default;
+
+  TovStar(double central_rest_mass_density, double polytropic_constant,
+          double polytropic_exponent) noexcept;
+
+  /// Retrieve a collection of variables at `(x, t)`
+  template <typename DataType, typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataType, 3>& x,
+                                         const double /*t*/,
+                                         tmpl::list<Tags...> /*meta*/) const
+      noexcept {
+    auto radial_vars =
+        radial_tov_solution().radial_variables(equation_of_state_, x);
+    return {get<Tags>(variables(x, tmpl::list<Tags>{}, radial_vars))...};
+  }
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& /*p*/) noexcept;  //  NOLINT
+
+  const EquationsOfState::PolytropicFluid<true>& equation_of_state() const
+      noexcept {
+    return equation_of_state_;
+  }
+
+ private:
+  friend bool operator==(const TovStar& lhs, const TovStar& rhs) noexcept {
+    // there is no comparison operator for the EoS, but should be okay as
+    // the `polytropic_exponent`s and `polytropic_constant`s are compared
+    return lhs.central_rest_mass_density_ == rhs.central_rest_mass_density_ and
+           lhs.polytropic_constant_ == rhs.polytropic_constant_ and
+           lhs.polytropic_exponent_ == rhs.polytropic_exponent_;
+  }
+
+  const RadialSolution& radial_tov_solution() const noexcept;
+
+  template <typename DataType>
+  auto variables(const tnsr::I<DataType, 3>& x,
```

Comment:
Should we maybe use the `Boost.preprocessor` code that was used in the upwind flux PR to generate all the forward decls? That'll make life easier in the future.